checks/fuzzing_test.go

```diff
@@ -0,0 +1,241 @@
+// Copyright 2020 Security Scorecard Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package checks
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+
+	"github.com/ossf/scorecard/v4/checker"
+	"github.com/ossf/scorecard/v4/clients"
+	mockrepo "github.com/ossf/scorecard/v4/clients/mockclients"
+)
+
+// Test_checkOSSFuzz is a test function for checkOSSFuzz.
+func Test_checkOSSFuzz(t *testing.T) {
+	t.Parallel()
+	//nolint
+	tests := []struct {
+		name        string
+		want        bool
+		response    clients.SearchResponse
+		wantErr     bool
+		wantFuzzErr bool
+	}{
+		{
+			name:     "Test_checkOSSFuzz failure",
+			want:     false,
+			response: clients.SearchResponse{},
+			wantErr:  false,
+		},
+		{
+			name: "Test_checkOSSFuzz success",
+			want: true,
+			response: clients.SearchResponse{
+				Hits: 1,
+			},
+			wantErr: false,
+		},
+		{
+			name:    "Test_checkOSSFuzz error",
+			want:    false,
+			wantErr: true,
+		},
+		{
+			name:        "Test_checkOSSFuzz fuzz error",
+			want:        false,
+			wantFuzzErr: true,
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			ctrl := gomock.NewController(t)
+
+			mockFuzz := mockrepo.NewMockRepoClient(ctrl)
+			mockFuzz.EXPECT().URI().Return("github.com/ossf/scorecard").AnyTimes()
+			mockFuzz.EXPECT().Search(gomock.Any()).
+				DoAndReturn(func(q clients.SearchRequest) (clients.SearchResponse, error) {
+					if tt.wantErr {
+						//nolint
+						return clients.SearchResponse{}, errors.New("error")
+					}
+					return tt.response, nil
+				}).AnyTimes()
+
+			req := checker.CheckRequest{
+				RepoClient:  mockFuzz,
+				OssFuzzRepo: mockFuzz,
+			}
+			if tt.wantFuzzErr {
+				req.OssFuzzRepo = nil
+			}
+
+			got, err := checkOSSFuzz(&req)
+
+			if (err != nil) != tt.wantErr {
+				t.Errorf("checkOSSFuzz() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if got != tt.want {
```

Comment:
can you add a verification of Warn/Info messages? (Same comment for other functions)

Reply:
Can you point me to an example of this?

Reply:
sure, see https://github.com/ossf/scorecard/blob/main/checks/dangerous_workflow_test.go#L160

Reply:
This would involve refactoring the original implementation for including the logging changes. Am I right? I don't see it in the `Fuzzing` check.

Reply:
I think in this case it would just be `NumberOfWarn: 0, NumberOfInfo: 0, NumberOfDebug: 0`, right?
Let me know if I'm mis-understanding the question.

Reply:
Sorry, what value does that add? I don't get it. Probably we should discuss this.

Reply:
Also the `dangerous_workflow.go` has reference to `checker.DetailLogger` but whereas `fuzzing.go` doesn't use that. So I don't see the reason. 

Probably I am missing the point.

Reply:
I mean that we want to know that no messages are being printed, and it should not change unexpectedly because it would affect the SARIF output for GitHub scanning dashboard. So if we ever add calls to Warn(), that's something we want to catch.

Reply:
re: no logger in Fuzzing(). logger is passed as a field in the checker request https://github.com/ossf/scorecard/blob/main/checker/check_request.go#L29. You just nee dto set the logger in the request instead of passing it as argument.

Reply:
Got it! Thanks for the details, it helped to understand it. But I am worried that this is going to make it harder for new devs to contribute to the project.